ui/src/main/webapp/src/app/analysis-context/select-applications.component.ts

```diff
@@ -0,0 +1,51 @@
+import {Component, Input, EventEmitter, Output} from "@angular/core";
+import {RegisteredApplication} from "windup-services";
+
+@Component({
+    selector: 'wu-select-applications',
+    templateUrl: './select-applications.component.html'
+})
+export class SelectApplicationsComponent {
+    @Input()
+    availableApps:RegisteredApplication[];
+    _selectedApps:RegisteredApplication[];
+
+    @Output()
+    selectedAppsChange:EventEmitter<RegisteredApplication[]> = new EventEmitter<RegisteredApplication[]>();
+
+    @Input()
+    set selectedApps(apps:RegisteredApplication[]) {
+        this._selectedApps = apps;
+        this.selectedAppsChange.emit(this._selectedApps);
```

Comment:
I'm surprised Angular is not complaining about this one. It is changing the value in @Input. Angular usually triggers some ugly exceptions like 

> item modified after it has been checked 

or something like that... Maybe it is intelligent enough not to complain, when you emit the same value as was the input one :)